pkg/apiserver/logsearch/router.go

```diff
@@ -0,0 +1,353 @@
+	endpoint.GET("/taskgroups/:id", s.TaskGroupGet)
+	endpoint.GET("/taskgroups/:id/preview", s.TaskGroupPreview)
+	endpoint.DELETE("/taskgroups/:id", s.TaskGroupDelete)
+}
+
+// @Summary List all tasks
+// @Description list all log search tasks
+// @Produce json
+// @Success 200 {array} TaskModel
+// @Failure 400 {object} httputil.HTTPError
+// @Router /logs/tasks [get]
+func (s *Service) TaskGetList(c *gin.Context) {
+	var tasks []TaskModel
+	err := s.db.Find(&tasks).Error
+	if err != nil {
+		httputil.NewError(c, http.StatusBadRequest, err)
+		return
+	}
+	c.JSON(http.StatusOK, tasks)
+}
+
+// @Summary Preview logs in a task
+// @Description preview fetched logs in a task by providing task ID
+// @Produce json
+// @Param id path string true "task id"
+// @Success 200 {array} PreviewModel
+// @Failure 400 {object} httputil.HTTPError
+// @Router /logs/tasks/{id}/preview [get]
+func (s *Service) TaskPreview(c *gin.Context) {
+	taskID := c.Param("id")
+	var lines []PreviewModel
+	err := s.db.Where("task_id = ?", taskID).Find(&lines).Error
+	if err != nil {
+		httputil.NewError(c, http.StatusBadRequest, err)
+		return
+	}
+	c.JSON(http.StatusOK, lines)
+}
+
+// @Summary Create a task group
+// @Description create and run a task group
+// @Produce json
+// @Success 200 {object} httputil.HTTPSuccess
+// @Failure 400 {object} httputil.HTTPError
+// @Failure 500 {object} httputil.HTTPError
+// @Router /logs/taskgroups [post]
+func (s *Service) TaskGroupCreate(c *gin.Context) {
+	// TODO: using parameters provided by client
+	endTime := time.Now().UnixNano() / int64(time.Millisecond)
+	startTime := int64(0)
+	var searchLogReq = &SearchLogRequest{
+		StartTime: startTime,
+		EndTime:   endTime,
+		Levels:    nil,
+		Patterns:  nil,
+	}
+	var components = []*Component{
+		{
+			ServerType: "tidb",
+			IP:         "127.0.0.1",
+			Port:       "4000",
+			StatusPort: "10080",
+		},
+	}
+	if s.scheduler.taskGroup != nil && s.scheduler.taskGroup.State == StateRunning {
+		httputil.NewError(c, http.StatusInternalServerError,
+			fmt.Errorf("cannot start, task group is %s", StateRunning))
+		return
+	}
+	s.scheduler.addTasks(components, searchLogReq)
+	err := s.scheduler.runTaskGroup(false)
+	if err != nil {
+		httputil.NewError(c, http.StatusInternalServerError, err)
+		return
+	}
+	httputil.Success(c)
+}
+
+// @Summary Download logs by task ID
+// @Description download logs by task id
+// @Produce application/zip
+// @Param id path string true "task id"
+// @Failure 400 {object} httputil.HTTPError
+// @Failure 500 {object} httputil.HTTPError
+// @Router /logs/tasks/{id}/download [get]
+func (s *Service) TaskDownload(c *gin.Context) {
+	taskID := c.Param("id")
+	task, err := s.queryTaskByID(taskID)
+	if err != nil {
+		httputil.NewError(c, http.StatusBadRequest, err)
+		return
+	}
+	f, err := os.Open(task.SavedPath)
+	if err != nil {
+		httputil.NewError(c, http.StatusInternalServerError, err)
+		return
+	}
+	_, err = io.Copy(c.Writer, f)
```

Comment:
Prefer to follow https://github.com/gin-gonic/gin#serving-data-from-reader . Also we'd better specify content-type and file name so that user is more easy to download.